app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectCreationActivity.kt

```diff
@@ -1,44 +1,54 @@
+
 package com.sdp13epfl2021.projmag
 
 import android.app.Activity
 import android.content.Intent
-import android.media.MediaFormat
 import android.net.Uri
-import android.opengl.Visibility
 import android.os.Bundle
 import android.provider.MediaStore
 import android.view.View
 import android.view.View.VISIBLE
 import android.widget.*
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.RecyclerView
 import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
-import com.google.firebase.storage.FirebaseStorage
-import com.sdp13epfl2021.projmag.database.FirebaseFileDatabase
+import com.sdp13epfl2021.projmag.activities.TagsSelectorActivity
 import com.sdp13epfl2021.projmag.database.ProjectUploader
 import com.sdp13epfl2021.projmag.database.Utils
 import com.sdp13epfl2021.projmag.model.ImmutableProject
 import com.sdp13epfl2021.projmag.model.Result
 import com.sdp13epfl2021.projmag.video.VideoSubtitlingActivity
 import com.sdp13epfl2021.projmag.video.VideoUtils
-import java.util.*
 
 
 const val FORM_TO_SUBTITLE_MESSAGE = "com.sdp13epfl2021.projmag.FROM_TO_SUBTITLE_MESSAGE"
 
 class Form : AppCompatActivity() {
+
     companion object {
         private const val REQUEST_VIDEO_ACCESS = 1
         private const val REQUEST_VIDEO_SUBTITLING = 2
+        private const val REQUEST_TAG_ACCESS = 3
     }
 
+
+    //tag selection related variables
+    private lateinit var tagRecyclerView: RecyclerView
+
+
+    //video related variables
+
     private var videoUri: Uri? = null
     private var subtitles: String? = null
 
+    private lateinit var listTags : Array<String>
```

Comment:
If we don't add tags, this will never be initialized.
Maybe an empty default Array could be better ini this case.

Reply:
fixed